zcash_client_sqlite/src/wallet/scanning.rs

```diff
@@ -791,46 +791,65 @@ pub(crate) mod tests {
     }
 
     pub(crate) fn test_with_nu5_birthday_offset<T: ShieldedPoolTester>(
+        initial_shard_blocks: u32,
         birthday_offset: u32,
         prior_block_hash: BlockHash,
+        insert_prior_roots: bool,
     ) -> (TestState<BlockCache>, T::Fvk, AccountBirthday, u32) {
         let st = TestBuilder::new()
             .with_block_cache()
-            .with_account_birthday(|rng, network, initial_chain_state| {
-                // We're constructing the birthday without adding any chain data.
-                assert!(initial_chain_state.is_none());
-
+            .with_initial_chain_state(|rng, network| {
                 // We set the Sapling and Orchard frontiers at the birthday height to be
                 // 1234 notes into the second shard.
                 let frontier_position = Position::from((0x1 << 16) + 1234);
-                let birthday_height =
-                    network.activation_height(NetworkUpgrade::Nu5).unwrap() + birthday_offset;
+                let initial_shard_end =
+                    network.activation_height(NetworkUpgrade::Nu5).unwrap() + initial_shard_blocks;
+                let birthday_height = initial_shard_end + birthday_offset;
```

Comment:
Document on `test_with_nu5_birthday_offset` that `birthday_offset` is an offset from `initial_shard_blocks`, as this is a semantic change to the function argument. Also document that `initial_shard_blocks` is an offset from NU5.